dev/src/generate/binding_python.rs

```diff
@@ -17,9 +17,87 @@
 
 use crate::generate::parser::Services;
 use anyhow::Result;
+use itertools::Itertools;
+use std::fs;
+use std::path::PathBuf;
 
-pub fn generate(services: &Services) -> Result<()> {
-    println!("{:?}", services);
+use super::parser::ConfigType;
+
+/// TODO: add a common utils to parse enabled features from cargo.toml
+fn enabled_service(srv: &str) -> bool {
+    match srv {
+        // not enabled in bindings/python/Cargo.toml
+        "etcd" | "foundationdb" | "ftp" | "hdfs" | "rocksdb" | "tikv" => false,
+        _ => true,
+    }
+}
+
+pub fn generate(project_root: PathBuf, services: &Services) -> Result<()> {
+    let mut s = fs::read_to_string(project_root.join("dev/templates/python"))
```

Comment:
Perhaps we could embed the templates here, so we won't need to edit multiple files whenever we want to modify the Python code.

Reply:
The other perspective also makes sense to me: moving all string construction logic into a template file.